distributed/worker.py

```diff
@@ -2098,8 +2098,7 @@ def transition_released_waiting(
             if dep_ts.state != "memory":
                 ts.waiting_for_data.add(dep_ts)
                 dep_ts.waiters.add(ts)
-                if dep_ts.state not in {"fetch", "flight"}:
-                    recommendations[dep_ts] = "fetch"
+                recommendations[dep_ts] = "fetch"
```

Comment:
fetch->fetch is a no-op.
flight->fetch is a no-op if not ts.done.

This aligns the logic here and in handle_acquire_replicas.